src/components/CowPenContextMenu/CowPenContextMenu.test.js

```diff
@@ -1,122 +1,158 @@
 import React from 'react'
-import { shallow } from 'enzyme'
-import Button from '@material-ui/core/Button'
+import { render, screen } from '@testing-library/react'
 
 import { generateCow } from '../../utils'
 import { PURCHASEABLE_COW_PENS } from '../../constants'
-import { cowColors } from '../../enums'
+import { cowColors, genders } from '../../enums'
 
 import {
   CowPenContextMenu,
   CowCard,
   CowCardSubheader,
 } from './CowPenContextMenu'
 
-let component
+jest.mock('../Item', () => ({
+  __esModule: true,
+  default: () => <></>,
+}))
 
 describe('CowPenContextMenu', () => {
-  beforeEach(() => {
-    component = shallow(
-      <CowPenContextMenu
-        {...{
-          cowBreedingPen: { cowId1: null, cowId2: null, daysUntilBirth: -1 },
-          cowForSale: generateCow(),
-          cowInventory: [],
-          debounced: {},
-          handleCowAutomaticHugChange: () => {},
-          handleCowBreedChange: () => {},
-          handleCowHugClick: () => {},
-          handleCowNameInputChange: () => {},
-          handleCowPurchaseClick: () => {},
-          handleCowSelect: () => {},
-          handleCowSellClick: () => {},
-          inventory: [],
-          money: 0,
-          purchasedCowPen: 1,
-          selectedCowId: '',
-        }}
-      />
-    )
-  })
+  let baseProps
 
   describe('cow selection', () => {
+    let testCow
+
+    beforeEach(() => {
+      jest.spyOn(Math, 'random').mockReturnValue(0)
+
+      baseProps = {
+        cowBreedingPen: { cowId1: null, cowId2: null, daysUntilBirth: -1 },
+        cowForSale: generateCow(),
+        cowInventory: [],
+        debounced: {},
+        handleCowAutomaticHugChange: () => {},
+        handleCowBreedChange: () => {},
+        handleCowHugClick: () => {},
+        handleCowNameInputChange: () => {},
+        handleCowPurchaseClick: () => {},
+        handleCowSelect: () => {},
+        handleCowSellClick: () => {},
+        inventory: [],
+        money: 0,
+        purchasedCowPen: 1,
+        selectedCowId: '',
+      }
+      testCow = generateCow({ id: 'foo' })
+    })
+
     describe('cow is not selected', () => {
       test('provides correct isSelected prop', () => {
-        component.setProps({
-          cowInventory: [generateCow({ id: 'foo' })],
-          selectedCowId: 'bar',
-        })
+        const { container } = render(
+          <CowPenContextMenu
+            {...{
+              ...baseProps,
+              cowInventory: [testCow],
+              selectedCowId: 'bar',
+            }}
+          />
+        )
 
-        expect(
-          component
-            .find('.card-list')
-            .find(CowCard)
-            .props().isSelected
-        ).toEqual(false)
+        expect(container).toMatchSnapshot()
```

Comment:
is there a specific rendered result we can check for instead? these snapshots that snap an entire rendered output are so sensitive to future changes i find they end up being a pain instead of helpful (literally every change breaks them). part of me think there's a chance this test isn't even needed with all the specific checks for the checkbox and heart statuses later on. and maybe the next snapshot too. what do you think? might be easier to chat in discord if you want!

Reply:
>is there a specific rendered result we can check for instead?

I scoped this in as closely as I could in https://github.com/jeremyckahn/farmhand/commit/f5f4f9dd450b17a9206885b32bcfb9d24fa432ea, but it's still pretty large, unfortunately.

>these snapshots that snap an entire rendered output are so sensitive to future changes i find they end up being a pain instead of helpful (literally every change breaks them).

I definitely agree, but short of checking that the `Card` component received the specific `raised` prop value (which is an antipattern worth moving away from), there's no great way that I can think of to check the rendered state of selected vs. non-selected `CowCard` state. To put it in concrete terms, this test is specifically interested in the classlists here: https://github.com/jeremyckahn/farmhand/blob/f5f4f9dd450b17a9206885b32bcfb9d24fa432ea/src/components/CowPenContextMenu/__snapshots__/CowPenContextMenu.test.js.snap#L5

and here: https://github.com/jeremyckahn/farmhand/blob/f5f4f9dd450b17a9206885b32bcfb9d24fa432ea/src/components/CowPenContextMenu/__snapshots__/CowPenContextMenu.test.js.snap#L371

The fact that the `MuiPaper-elevationX` classes are the sole distinguishing characteristic of these two variations is a Material-UI implementation detail that our application tests should not be concerned with. The fact that there are hundreds of lines of unrelated noise are an unfortunate side effect of how React Testing Library rendering and Jest snapshots work.

I would _love_ a better solution here that is simpler, more robust, and adherent to best practices. Do you have any ideas that might meet all these needs better?

>there's a chance this test isn't even needed with all the specific checks for the checkbox and heart statuses later on. and maybe the next snapshot too.

I think it's worth testing that our component renders selected vs. non-selected `CowCard`s correctly. I suppose a lighter-weight approach to doing this is to conditionally set a `selected` class on the `CowCard`, but then we're adding functionality specifically for the sake of testing.

---------------------

To sum up, I can't find a solution that feels great here, but what's here feels most closely aligned to React and testing best practices. What are your thoughts about this?

Reply:
in this case the long term solution may be that we need to add more to distinguish the two states, the className alone is quite meaningless. i'm cool with future improvements rather than holding up this PR longer, but something like rendering some [visually hidden](https://css-tricks.com/inclusively-hidden/) text saying "selected" for the selected one would be an option.

Reply:
>something like rendering some visually hidden text saying "selected" for the selected one would be an option.

That's a great idea! I think I'll roll that into this PR before merging because it solves this problem well and also adds a meaningful user feature for little work. Thanks for the suggestion!

Reply:
This is addressed in https://github.com/jeremyckahn/farmhand/commit/d7e793e3f426a56962ff9c2ce2c6b60ab29eb7df. This actually cleans up this whole PR quite nicely. Thanks again for the great suggestion!